File: SnowForecast/app/src/main/java/com/sners/snowforecast/data/MinutelyData.java
package com.sners.snowforecast.data;

import com.sners.snowforecast.data.IntervalData;
import com.sners.snowforecast.data.WeatherConstants;
import com.sners.snowforecast.data.WeatherConstants;


import org.json.JSONException;
import org.json.JSONObject;

public class MinutelyData extends IntervalData {



	////////////////////////////////////////////////////////////////////////////////
	//	Constructor
	////////////////////////////////////////////////////////////////////////////////
	public MinutelyData(JSONObject jsonMinutely)
	{
		try {
			time 				= jsonMinutely.getString(WeatherConstants.TIME_LOCAL);
			precipIntensity 	= (float)jsonMinutely.getDouble(WeatherConstants.PRECIP_INTENSITY);
			precipProbability 	= (float)jsonMinutely.getDouble(WeatherConstants.PRECIP_TYPE_SNOW);
			if (precipProbability > 100) {
				precipProbability = 100;
			}


			if (precipProbability > 0) {
				float snow 	= (float)jsonMinutely.getDouble(WeatherConstants.PRECIP_TYPE_SNOW);
				if (snow > 0) {
					weatherWords.add(WeatherConstants.PRECIP_TYPE_SNOW);
				}
				else {
					weatherWords.add(WeatherConstants.PRECIP_TYPE_RAIN);
				}

			}
			//	We have a precip and a snow - there's nothing else
			//	Still need to decide what's going on that graph
			//precipType 			= jsonMinutely.getInt(WeatherConstants.PRECIP_TYPE);;
			//weatherCode			= jsonMinutely.getInt(WeatherConstants.WEATHER_CODE);


		} catch (JSONException e) {
			e.printStackTrace();
		}

	}

}

File: SnowForecast/app/src/main/java/com/sners/snowforecast/data/MinutelyData.kt
package com.sners.snowforecast.data

import com.sners.snowforecast.weather.WeatherConstants
import org.json.JSONException
import org.json.JSONObject

/**
 * A data class of minutely weather conditions
 * It takes the raw json and converts it into a data structure
 * @param jsonMinutely The raw current data coming in from the API call
 */
class MinutelyData(jsonMinutely: JSONObject) : IntervalData() {

    /**
     * localTime Time for this minute instance
     */
    var localTime = "";

    /**
     * utcTime Time for this minute instance
     */
    var utcTime = "";

    /**
     * unixTime Time for this minute instance
     */
    var unixTime = 0L;


    /*
  Constructor
   */
    init {
        try {
            time = jsonMinutely.getString(WeatherConstants.TIME_LOCAL)

            localTime = jsonMinutely.getString(WeatherConstants.TIMESTAMP_LOCAL)
            utcTime = jsonMinutely.getString(WeatherConstants.TIMESTAMP_UTC)
            unixTime = jsonMinutely.getLong(WeatherConstants.TIMESTAMP)
            precipIntensity = jsonMinutely.getDouble(WeatherConstants.PRECIP_INTENSITY).toFloat() / 24f
            precipProbability = jsonFloatValueFor(WeatherConstants.PRECIP_TYPE_SNOW, jsonMinutely) ?: 0f
            if (precipProbability > 100) {
                precipProbability = 100f
            }
            val snow = jsonFloatValueFor(WeatherConstants.PRECIP_TYPE_SNOW, jsonMinutely) ?: 0f
            if (snow > 0) {
                weatherWords.add(WeatherConstants.PRECIP_TYPE_SNOW)
            } else if (precipIntensity > 0) {
                weatherWords.add(WeatherConstants.PRECIP_TYPE_RAIN)
            }
        } catch (e: JSONException) {
            e.printStackTrace()
        }
    }
}